### PR TITLE
SITL: fix pthread_setname_np to build on mac

### LINKED ISSUE #7178: SITL build fails, fatal error no function "pthread_setname_np"
#### Issue details
SITL build with sim_vehicle.py fails

I'm using this command:
`sim_vehicle.py -v ArduCopter -f quad --no-mavproxy`

#### Version
Latest version on this repository, https://github.com/ArduPilot/ardupilot/commit/c9652fec3ccd957e1082dc9f1f27ea7bd419be9f

#### Platform
[  ] All
[  ] AntennaTracker
[x] Copter *(maybe others too?)*
[  ] Plane
[  ] Rover
[  ] Submarine

#### Airframe type
quad *(maybe others too?)*

#### Hardware type
None, SITL on Mac OSX

#### Logs

```
SIM_VEHICLE: Building
SIM_VEHICLE: "ardupilot/modules/waf/waf-light" "build" "--target" "bin/arducopter"
Waf: Entering directory `ardupilot/build/sitl'
[458/828] Compiling libraries/AP_AccelCal/AP_AccelCal.cpp
[467/828] Compiling libraries/SITL/SIM_FlightAxis.cpp
../../libraries/SITL/SIM_FlightAxis.cpp:93:5: fatal error: no matching function for call to 'pthread_setname_np'
    pthread_setname_np(pthread_self(), "ardupilot-flightaxis");
    ^~~~~~~~~~~~~~~~~~
/usr/include/pthread.h:475:5: note: candidate function not viable: requires 1 argument, but 2 were provided
int     pthread_setname_np(const char*);
        ^
1 error generated.

[825/828] Compiling libraries/SITL/SIM_Plane.cpp
[826/828] Compiling libraries/SITL/SIM_Sprayer.cpp
Waf: Leaving directory `ardupilot/build/sitl'
Build failed
 -> task in 'objs/SITL' failed (exit status 1): 
	{task 4568049536: cxx SIM_FlightAxis.cpp -> SIM_FlightAxis.cpp.0.o}
['/usr/bin/clang++', '-std=gnu++11', '-fdata-sections', '-ffunction-sections', '-fno-exceptions', '-fsigned-char', '-Wall', '-Wextra', '-Wformat', '-Wshadow', '-Wpointer-arith', '-Wcast-align', '-Wundef', '-Wno-unused-parameter', '-Wno-missing-field-initializers', '-Wno-reorder', '-Wno-redundant-decls', '-Wno-unknown-pragmas', '-Werror=format-security', '-Werror=array-bounds', '-Werror=uninitialized', '-Werror=init-self', '-Werror=switch', '-Wfatal-errors', '-fcolor-diagnostics', '-Wno-gnu-designator', '-Wno-inconsistent-missing-override', '-Wno-mismatched-tags', '-Wno-gnu-variable-sized-type-not-at-end', '-Wno-c++11-narrowing', '-O3', '-include', 'ap_config.h', '-Ilibraries', '-Ilibraries/GCS_MAVLink', '-I.', '-I../../libraries', '-I../../libraries/AP_Common/missing', '-DSKETCHBOOK="ardupilot"', '-DCONFIG_HAL_BOARD=HAL_BOARD_SITL', '-DCONFIG_HAL_BOARD_SUBTYPE=HAL_BOARD_SUBTYPE_NONE', '../../libraries/SITL/SIM_FlightAxis.cpp', '-c', '-oardupilot/build/sitl/libraries/SITL/SIM_FlightAxis.cpp.0.o']
SIM_VEHICLE: Build failed
```

### PR DESCRIPTION
**Pull request by guglie**
This fixes #7178 (SITL build failing)

### CODE REVIEW

File: libraries/SITL/SIM_FlightAxis.cpp [hunk 1]
```diff
@@ -104,7 +104,11 @@ void *FlightAxis::update_thread(void *arg)
     FlightAxis *flightaxis = (FlightAxis *)arg;
 
 #ifndef __CYGWIN__
```
Comment (peterbarker) [inline libraries/SITL/SIM_FlightAxis.cpp]: Could we simplify this with `#ifdef __CYGWIN__` and `#elif`?

Comment (guglie) [reply libraries/SITL/SIM_FlightAxis.cpp]: Do you mean like this?
```
#ifdef __CYGWIN__
    //do nothing
#elif defined(__APPLE__) && defined(__MACH__)
    pthread_setname_np("ardupilot-flightaxis");
#else
    pthread_setname_np(pthread_self(), "ardupilot-flightaxis");
#endif
```

Comment (peterbarker) [reply libraries/SITL/SIM_FlightAxis.cpp]: Yes, indeed.  I think that's downright pretty compared to the nested-if.  As pretty as preprocessor stuff gets, anyway.

Only possible thing I could see is explaining why cygwin doesn't do the setname.

Your thoughts?

Comment (guglie) [reply libraries/SITL/SIM_FlightAxis.cpp]: Yes, seems more readable.

I don't know why cygwin doesn't support it, that was a fix pushed by @tridge in cdd350f93e55df764d2f0bc7eaa4f94ed5fa7456.

After a bit of research it just seems unsupported, but I found some interesting info in both answers here https://stackoverflow.com/questions/30346649 and also here https://stackoverflow.com/a/7989973/648044

**Comment by guglie** [comment]
Updated 😃 

---

lost in comments to previous edit:
`pthread_setname_np()` seems just unsupported in Cygwin

https://stackoverflow.com/questions/30346649/

https://stackoverflow.com/questions/2369738/can-i-set-the-name-of-a-thread-in-pthreads-linux/7989973#7989973

**Comment by peterbarker** [comment]
Merged, thanks.

**Pull request merged**